src/Tickets/Commerce/Gateways/Square/Webhooks.php

```diff
@@ -0,0 +1,855 @@
+use TEC\Tickets\Commerce\Gateways\Square\WhoDat;
+use Tribe__Utils__Array as Arr;
+
+use TEC\Common\Contracts\Provider\Controller as Controller_Contract;
+/**
+ * Class Webhooks
+ *
+ * @since TBD
+ *
+ * @package TEC\Tickets\Commerce\Gateways\Square
+ */
+class Webhooks extends Controller_Contract {
+
+	/**
+	 * Option key for storing webhook signatures.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public static $option_webhook_signature = 'tickets-commerce-square-webhook-signature';
+
+	/**
+	 * Option key for storing webhook IDs.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public static $option_webhook_id = 'tickets-commerce-square-webhook-id';
+
+	/**
+	 * Option key for storing webhook last check status.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public static $option_webhook_last_check = 'tickets-commerce-square-webhook-last-check';
+
+	/**
+	 * Option key for storing webhook configuration status.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public static $option_webhook_configuration = 'tickets-commerce-square-webhook-configuration';
+
+	/**
+	 * Option key for storing available event types.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public static $option_available_event_types = 'tickets-commerce-square-available-event-types';
+
+	/**
+	 * Square API version used for webhooks.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected $api_version = '2023-12-13';
+
+	/**
+	 * Event types to subscribe to.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	protected $event_types = [
+		'payment.created',
+		'payment.updated',
+		'refund.created',
+		'refund.updated',
+	];
+
+	/**
+	 * Register the service provider.
+	 *
+	 * @since TBD
+	 */
+	public function do_register(): void {
+		// Add cron event for webhook health check
+		add_action( 'init', [ $this, 'register_cron_events' ] );
+		add_action( 'tec_tickets_commerce_square_check_webhooks', [ $this, 'check_webhook_health' ] );
+
+		// Add AJAX handler for webhook registration
+		add_action( 'wp_ajax_tec_tickets_commerce_square_register_webhook', [ $this, 'ajax_register_webhook' ] );
+
+		// Run initial checks for admin pages
+		add_action( 'admin_init', [ $this, 'maybe_run_initial_checks' ] );
+	}
+
+	/**
+	 * Unregister hooks and cleanup.
+	 *
+	 * @since TBD
+	 */
+	public function unregister(): void {
+		remove_action( 'init', [ $this, 'register_cron_events' ] );
+		remove_action( 'tec_tickets_commerce_square_check_webhooks', [ $this, 'check_webhook_health' ] );
+		remove_action( 'admin_init', [ $this, 'maybe_run_initial_checks' ] );
+		wp_clear_scheduled_hook( 'tec_tickets_commerce_square_check_webhooks' );
+	}
+
+	/**
+	 * Register cron events for webhook health check.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function register_cron_events() {
+		if ( ! wp_next_scheduled( 'tec_tickets_commerce_square_check_webhooks' ) ) {
+			wp_schedule_event( time(), 'daily', 'tec_tickets_commerce_square_check_webhooks' );
+		}
```

Comment:
Why would we do wp corn when our plugins come packaged with AS ? AS is far superior to wp's native cron...

Let's please replace with AS functions instead